api/src/i18n/repositories/language.repository.ts

```diff
@@ -43,14 +43,17 @@ export class LanguageRepository extends BaseRepository<
       Document<Language, any, any>,
       unknown,
       Language,
-      'deleteOne' | 'deleteMany'
+      'deleteOne'
```

Comment:
Let's remove this function and use OnEvent('hook:language:preDelete') in the NlpSampleService.
const languages = this.languageService.find(criteria)
await this.nlpSampleService.updateMany(criteria, { $set:{ language: null } })